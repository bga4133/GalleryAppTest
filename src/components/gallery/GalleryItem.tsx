import React from "react";

interface PropTypes {
  previewUrl: string;
  previewId: string;
}

export const GalleryItem = ({ previewUrl, previewId }: PropTypes) => {
  return (
    <>
      <img
        className="panel__gallery__img"
        key={`photo${previewUrl}`}
        src={previewUrl}
        alt={previewId}
      />
    </>
  );
};
