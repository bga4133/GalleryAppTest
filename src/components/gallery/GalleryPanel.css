.panel__gallery {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 10px;
  justify-content: center;
  width: 100%;
  min-width: 100%;
}

.panel__gallery__img {
  border-radius: 10px;
  display: flex;
  flex-grow: 1;
  flex-shrink: 1;
}

.no__content {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

.no__content__title {
  color: rgb(226, 200, 0);
}

@media screen and (max-width: 1200px) {
  .panel__gallery__img {
    max-width: 188px;
  }
}

@media screen and (max-width: 802px) {
  .panel__gallery__img {
    max-width: 160px;
  }
}

@media screen and (max-width: 600px) {
  .panel__gallery {
    gap: 0px;
  }

  .panel__gallery__img {
    max-width: 100px;
    border-radius: 0px;
  }
}

@media screen and (max-width: 360px) {
  .panel__gallery__img {
    max-width: 90px;
  }
}
