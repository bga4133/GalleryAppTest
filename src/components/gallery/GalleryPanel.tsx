import React, { ReactComponentElement, useState, useEffect } from "react";
import "./GalleryPanel.css";
import { photosApi, GetPhotosResponse } from "../../__dont_modify__/api/photos";
import { Photo } from "../../types/photos";
import { ApiError } from "../ApiError/ApiError";
import { GalleryItem } from "./GalleryItem";
import { LoadingModal } from "../shared/LoadingModal/LoadingModal";

const { getPhotos } = photosApi;
export const GalleryPanel: () => ReactComponentElement<any> = () => {
  // states
  const [photos, setPhotos] = useState<Photo[] | null>([]);
  const [isLoading, setIsLoading] = useState(false);

  const getPhotosFromApi = async () => {
    setIsLoading(true);
    try {
      const { photos }: GetPhotosResponse = await getPhotos({ page: 1 });
      setPhotos(photos);
      setIsLoading(false);
    } catch (error) {
      setPhotos(null);
      setIsLoading(false);
      throw new Error("Obbsss error");
    }
  };

  useEffect(() => {
    getPhotosFromApi();
  }, []);
  return (
    <>
      <div className="panel__gallery" data-testid="imgId">
        {photos?.map(({ previewUrl, previewId, id }: Photo) => (
          <GalleryItem
            key={`photo${previewUrl}`}
            previewUrl={previewUrl}
            previewId={previewId}
          />
        ))}
        {photos == null && <ApiError />}
        {isLoading && <LoadingModal />}
      </div>
    </>
  );
};
