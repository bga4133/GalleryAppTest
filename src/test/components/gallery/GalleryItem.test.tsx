import { GalleryItem } from "../../../components/gallery/GalleryItem";
import { render, screen, waitFor } from "@testing-library/react";

describe("test in <GalleryItem />", () => {
  const mockedPhoto = {
    previewId: "1664898372517",
    previewUrl: "https://picsum.photos/300/300?random=0.3833849915413208"
  };
  it("render the component", async () => {
    const { container } = render(
      <GalleryItem
        previewUrl={mockedPhoto.previewUrl}
        previewId={mockedPhoto.previewId}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("should show the correct src", async () => {
    const { container } = render(
      <GalleryItem
        previewUrl={mockedPhoto.previewUrl}
        previewId={mockedPhoto.previewId}
      />
    );
    const img = container.querySelector("img");
    expect(img).toBeTruthy();
    expect(img!.src).toBe(mockedPhoto.previewUrl);
  });
});
