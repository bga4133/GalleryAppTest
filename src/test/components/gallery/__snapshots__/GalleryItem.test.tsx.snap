// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GalleryItem /> render the component 1`] = `
<div>
  <img
    alt="1664898372517"
    class="panel__gallery__img"
    src="https://picsum.photos/300/300?random=0.3833849915413208"
  />
</div>
`;
